spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -326,6 +344,96 @@ private static FlatMapFunction<Iterator<String>, String> listDirsRecursively(
     };
   }
 
+  @VisibleForTesting
+  static List<String> findOrphanFiles(SparkSession spark,
+                                      Dataset<Row> actualFileDF,
+                                      Dataset<Row> validFileDF,
+                                      Map<String, String> equalSchemes,
+                                      Map<String, String> equalAuthorities,
+                                      PrefixMismatchMode prefixMismatchMode) {
+    Map<String, String> equalSchemesMap = flattenMap(equalSchemes);
+    Map<String, String> equalAuthoritiesMap = flattenMap(equalAuthorities);
+
+    Dataset<PathProxy> normalizedActualFileDF = actualFileDF.mapPartitions(
+        toFileMetadata(equalSchemesMap, equalAuthoritiesMap),
+        Encoders.bean(PathProxy.class)).as("actual");
+    Dataset<PathProxy> normalizedValidFileDF = validFileDF.mapPartitions(
+        toFileMetadata(equalSchemesMap, equalAuthoritiesMap),
+        Encoders.bean(PathProxy.class)).as("valid");
+
+    Column actualFileName = normalizedActualFileDF.col("path");
+    Column validFileName = normalizedValidFileDF.col("path");
+
+    SetAccumulator<Pair<String, String>> setAccumulator = new SetAccumulator<>();
+    spark.sparkContext().register(setAccumulator);
+
+    List<String> orphanFiles = normalizedActualFileDF.joinWith(normalizedValidFileDF,
+            actualFileName.equalTo(validFileName), "leftouter")
+        .mapPartitions(findOrphanFilesMapPartitions(prefixMismatchMode, setAccumulator), Encoders.STRING())
+        .collectAsList();
+
+    if (prefixMismatchMode == PrefixMismatchMode.ERROR && !setAccumulator.value().isEmpty()) {
+      throw new ValidationException("Unable to deterministically find all orphan files." +
```

Comment:
What about giving more details and instructions to the user? Something like this?

```
throw new ValidationException("Unable to determine whether certain files are orphan. " +
    "Metadata references files that match listed/provided files except for authority/scheme. " +
    "Please, inspect the conflicting authorities/schemes and provide which of them are equal " +
    "by further configuring the action via equalSchemes() and equalAuthorities() methods. " +
    "Set the prefix mismatch mode to 'NONE' to ignore remaining locations with conflicting " +
    "authorities/schemes or to 'DELETE' iff you are ABSOLUTELY confident that remaining conflicting " +
    "authorities/schemes are different. It will be impossible to recover deleted files. " +
    "Conflicting authorities/schemes: %s.", conflicts.value());
```

Reply:
done